SDK-v4/gdpr/app/src/main/java/com/localytics/gdpr/LocalyticsGDPRWrapper.java

```diff
@@ -0,0 +1,246 @@
+package com.localytics.gdpr;
+
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+
+import com.localytics.android.Localytics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This class acts as a wrapper around the Localytics implementation to ensure that no data can be
+ * passed to the Localytics system if a user has opted out of data collection.
+ *
+ * This implementation expects that every time the app is created, the opted in state of the user is
+ * checked against a backend source of truth.  As a result, the opted in state will be stored as an
+ * instance variable, and persistent storage will not be required.
+ */
+public class LocalyticsGDPRWrapper {
+
+    private static LocalyticsGDPRWrapper INSTANCE = null;
+
+    /**
+     * A boolean representing the user's current opt out status
+     */
+    private boolean optedOut = false;
+
+    /**
+     * A boolean representing whether or not datapoints can immediately be passed to Localytics.
+     * This value should be true when the current user's opt in status is unknown by the app.
+     */
+    private boolean queueDatapoints = false;
+
+    /**
+     * A queue of event datapoints that have been triggered by the app while the user's opt in state is unknown
+     */
+    private List<Runnable> eventQueue = new ArrayList<>();
+
+    static LocalyticsGDPRWrapper getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new LocalyticsGDPRWrapper();
+        }
+        return INSTANCE;
+    }
+
+    private LocalyticsGDPRWrapper() {
+        //private constructor to ensure getInstance is called.
+    }
+
+    void setLoggingEnabled(boolean loggingEnabled) {
+        Localytics.setLoggingEnabled(loggingEnabled);
+    }
+
+    //    Integration/Session tracking methods.
+
+    public void autoIntegrate(Application application) {
+        throw new RuntimeException("You can't use autoIntegrate while supporting GDPR");
+    }
+
+    public void integrate(Context context) {
```

Comment:
You'll actually need to pass in the app key here, correct? And the `localytics.xml` should be removed as Anand pointed out

Reply:
The customer can just include the app key in the "wrappper" call - same as I did here.

Reply:
I see what you're saying... I had totally confused myself